hadoop-ozone/recon/src/main/java/org/apache/hadoop/ozone/recon/fsck/ContainerHealthTask.java

```diff
@@ -435,6 +465,9 @@ public static boolean retainOrUpdateRecord(
       case MISSING:
         returnValue = container.isMissing() && !container.isEmpty();
         break;
+      case EMPTY_MISSING:
```

Comment:
Thanks for pointing this out @devmadhuu 

This change of Negative Sized was added by this :- https://github.com/apache/ozone/pull/6178
The Negative sized containers do get added to the table, but they do have a bug in their implementation :- 

### Iteration 1 of `ContainerHealthTask`:

- **What happens**: The `checkAndProcessContainers` method fetches all containers, including the ones with negative sizes, since `processedContainers` is empty initially.
- **Result**: The negative-sized containers are processed and added to the `UNHEALTHY_CONTAINERS` table by `handleNegativeSizedContainers`.

### Iteration 2:

- **What happens**: The `processExistingDBRecords` method processes the previously added negative-sized containers. When it checks the records in the table, `retainOrUpdateRecord` returns `false` for negative-sized containers because the logic for keeping negative-sized records isn't present.
- **Result**: The negative-sized containers are deleted from the `UNHEALTHY_CONTAINERS` table, and the container is added to `processedContainers`. As a result, `checkAndProcessContainers` skips this container because it’s in the `processedContainers` set.

### Iteration 3:

- **What happens**: In this iteration, since there are no records for negative-sized containers in the `UNHEALTHY_CONTAINERS` table (due to their deletion in the previous iteration), the `processExistingDBRecords` method does not delete anything.
- **Result**: The negative-sized containers are not present in the table, so `checkAndProcessContainers` processes them again and adds them back to the `UNHEALTHY_CONTAINERS` table.

### Iteration 4:

- **What happens**: The process from Iteration 2 is repeated, where the `processExistingDBRecords` method deletes the negative-sized containers again because `retainOrUpdateRecord` still returns `false`.
- **Result**: The container gets deleted again, and the loop continues.

### Conclusion:

This behaviour creates a **cycle** where the negative-sized containers are added in one iteration and deleted in the next. This is due to the absence of logic in `generateUnhealthyRecords` to retain negative-sized containers. Adding proper handling for negative-sized containers in `generateUnhealthyRecords` should prevent this cycle.

Reply:
Thanks @ArafatKhan2198  for your explanation above. My point is do we need to insert `NEGATIVE_SIZE` and `EMPTY_MISSING` containers in `UnhealthyContainers` table because they are anyway getting deleted in next iteration. There can be two ways to handle this:

1. Either DO NOT insert both these types containers in `UnhealthyContainers` table because they are getting deleted anyways in next iteration and we have logs to know the information on current count of both these types of containers with each task run. You can check the correctness of information in logs. This will make things simpler because no need to take care of constraint violation exception in case user moves to next higher version. And any existing MISSING type old containers which have actually zero keys needs to be cleaned up, so that Recon will not report such empty missing containers.
2. Or, Insert both these types of containers in `UnhealthyContainers` table which makes things little complex in terms of management of this data which is not being shown over Recon UI. Because such containers will continue to be lying in `UnhealthyContainers` table unless until we introduce some mechanism to clean up such containers periodically and then when and how to do cleanup of such containers makes things little overly complex.

By having solution 1, we also need to make sure about not to miss the corner edge case of updating container states of such containers after verification from SCM.

Reply:
I agree @devmadhuu with the first solution! Based on my experience with customer issues, backward compatibility can be a significant challenge. We are anyways printing out the logs for it so no point in storing them.

Reply:
@ArafatKhan2198 @devmadhuu 
There is one more changes done, rename of one filed ALL_REPLICAS_BAD which also can cause compatibility issue.
IMO, current we can re-create constraint dropping and re-creating for this.

But we need support Schema upgrade feature for compatibility scenario for this DB.